src/Eloquent/Builder.php

```diff
@@ -56,6 +56,18 @@ class Builder extends EloquentBuilder
         'tomql',
     ];
 
+    /**
+     * @return ($function === null ? PipelineBuilder : self)
+     *
+     * @inheritdoc
+     */
+    public function aggregate($function = null, $columns = ['*'])
+    {
+        $result = $this->toBase()->aggregate($function, $columns);
```

Comment:
Also, I have no idea what actually causes `toBase()` to return `MongoDB\Laravel\Query\Builder` instead of the Eloquent object. Best I can tell, `toBase()` is not overridden so I'd expect this to invoke [`Illuminate\Database\Eloquent\Builder::toBase()`](https://github.com/illuminate/database/blob/v11.0.6/Eloquent/Builder.php#L1760).

Reply:
Without this change, the call stack is:

- [AggregationBuilderTest.php:39](https://github.com/GromNaN/laravel-mongodb-fork/blob/05054e5491f22a080a0d75acb03ad142b37f2a08/tests/Query/AggregationBuilderTest.php#L39), `MongoDB\Laravel\Tests\Query\AggregationBuilderTest->testCreateFromQueryBuilder()`
- [Builder.php:1974](https://github.com/laravel/framework/blob/8af24b819e55beaf440954516f926d0f5f9abafd/src/Illuminate/Database/Eloquent/Builder.php#L1974), `Illuminate\Database\Eloquent\Builder->__call()` => Always returns the `Illuminate\Database\Eloquent\Builder`
- [ForwardsCalls.php:23](https://github.com/laravel/framework/blob/8af24b819e55beaf440954516f926d0f5f9abafd/src/Illuminate/Support/Traits/ForwardsCalls.php#L23), `Illuminate\Database\Eloquent\Builder->forwardCallTo()`
- [Builder.php:602](https://github.com/GromNaN/laravel-mongodb-fork/blob/05054e5491f22a080a0d75acb03ad142b37f2a08/src/Query/Builder.php#L603), `MongoDB\Laravel\Query\Builder->aggregate()`

There is several methods in `Illuminate\Database\Eloquent\Builder` that a calls the `toBase` query builder and pass the result of the query.